app/routes/docs_.schema.tsx

```diff
@@ -39,26 +43,46 @@ export async function loader({
 
 export default function Schema() {
   const { version, '*': path } = useParams()
+  if (!path) throw new Error('Path is not defined.')
 
   const [showVersions, setShowVersions] = useState(false)
   const { versions } = useLoaderData()
+  const windowSize = useWindowSize()
+  const isSchema = path.endsWith('.schema.json')
 
+  const splitPathArray: string[] = path.split('/')
+  const breadCrumbs = splitPathArray.map((item, index) => {
+    const crumbPath: string = splitPathArray.slice(0, index + 1).join('/')
+    return (
+      <Breadcrumb key={item} current={index + 1 == splitPathArray.length}>
+        {index + 1 == splitPathArray.length ? (
+          <h2 className="margin-y-0 display-inline">
+            {item.replace('.schema.json', '')}
+          </h2>
+        ) : (
+          <BreadcrumbLink href={`/docs/schema/${version}/${crumbPath}`}>
+            <span>{item}</span>
+          </BreadcrumbLink>
+        )}
+      </Breadcrumb>
+    )
+  })
```

Comment:
- [ ] I think that you can use spread syntax here so that you can avoid the comparisons.
- [ ] You might as well have this return the breadcrumb bar itself rather than a list of components, so that this is a proper React component.

```jsx
const [...breadcrumbs, lastBreadcrumb] = path.split('/')
return (
  <BreadcrumbBar>
    {
      breadcrumbs.map(
        (breadcrumb) => <Breadcrumb>fill me in for path components</Breadcrumb>
      )
    }
   <Breadcrumb>fill me in for filename</Breadcrumb>
  </BreadcrumBar>
)
```

Reply:
I think I got this idea working on my latest commit